public/app/plugins/panel/timeseries/plugins/ThresholdControlsPlugin.tsx

```diff
@@ -42,7 +42,7 @@ export const ThresholdControlsPlugin: React.FC<ThresholdControlsPluginProps> = (
     if (!thresholds) {
       return null;
     }
-    const scale = buildScaleKey(fieldConfig as FieldConfig<GraphFieldConfig>);
+    const scale = buildScaleKey(fieldConfig.defaults as FieldConfig<GraphFieldConfig>);
```

Comment:
We should be able to drop the type casting here, since `fieldConfig.defaults` is a proper `FieldConfig` and not a `FieldConfigSource`.

```suggestion
    const scale = buildScaleKey(fieldConfig.defaults);
```